File: app/src/main/java/com/example/madworkshop/Model/DrinkRepository.java
package com.example.madworkshop.Model;

import android.app.Application;
import androidx.lifecycle.LiveData;
import java.util.List;

public class DrinkRepository {
    //We create our drink dao and a list of drinks
    private DrinkDao mDrinkDao;
    private LiveData<List<Drink>> mAllDrinks;

    //The constructor for the repository
    public DrinkRepository(Application application) {
        //We get the database from our Database class
        DrinkRoomDatabase db = DrinkRoomDatabase.getDatabase(application);
        //We instantiate our drinkDatabase
        mDrinkDao = db.drinkDao();
    }

    public LiveData<List<Drink>> getAllDrinks() {
        //Get all the drinks in our database and then return them.
        mAllDrinks = mDrinkDao.getAllDrinks();
        return mAllDrinks;
    }

    public void insertDrink(Drink drink) {
        //Insert a new drink into our database
        DrinkRoomDatabase.databaseWriteExecutor .execute(() -> {
            mDrinkDao.insertDrink(drink);
        });
    }
}

File: app/src/main/java/com/example/madworkshop/Model/DrinkRepository.kt
package com.example.madworkshop.Model

import android.app.Application
import androidx.lifecycle.LiveData

class DrinkRepository(application: Application?) {
    //We create our drink dao and a list of drinks
    private var mDrinkDao: DrinkDao? = null
    private var mAllDrinks: LiveData<List<Drink?>?>? = null

    //The constructor for the repository
    init {
        //We get the database from our Database class
        val db = application?.let { DrinkRoomDatabase.getDatabase(it) }
        //We instantiate our drinkDatabase
        if (db != null) {
            mDrinkDao = db.drinkDao()
        }
    }

    val allDrinks: LiveData<List<Drink?>?>?
        get() {
            //Get all the drinks in our database and then return them.
            mAllDrinks = mDrinkDao?.allDrinks
            return mAllDrinks
        }

    fun insertDrink(drink: Drink?) {
        //Insert a new drink into our database
        DrinkRoomDatabase.databaseWriteExecutor.execute { mDrinkDao?.insertDrink(drink) }
    }
}
